checker/src/path.rs

```diff
@@ -744,6 +744,18 @@ impl PathRefinement for Rc<Path> {
                     qualifier.refine_parameters(arguments, result, pre_environment, fresh);
                 let refined_selector =
                     selector.refine_parameters(arguments, result, pre_environment, fresh);
+                // todo: more path canonicalization?
```

Comment:
Since refine_parameters is always followed by refine_paths, it seemed reasonable to make refine_parameters as special purpose as possible and not do anything that refine_paths does. Just looking at this in isolation it not quite clear to me that this should not instead happen in refine_paths. I kind of expect that to already be the case, so things might be a bit more complicated than they seem. For example, *&path and path are not the same thing if path refers to a structure, because the * effectively copies the structure in that case.

Reply:
I see. I was looking at the intermediate values between `refine_parameters` and `refine_paths` and got confused. Reverted these changes.